src/PowerFlowData.jl

```diff
@@ -256,6 +285,95 @@ get_calculate_loss_factors(pfd::PowerFlowData) = pfd.calculate_loss_factors
 get_voltage_stability_factors(pfd::PowerFlowData) = pfd.voltage_stability_factors
 get_calculate_voltage_stability_factors(pfd::PowerFlowData) =
     pfd.calculate_voltage_stability_factors
+get_lcc_p_set(pfd::PowerFlowData) = pfd.lcc.p_set
+get_lcc_dc_line_resistance(pfd::PowerFlowData) = pfd.lcc.dc_line_resistance
+get_lcc_rectifier_tap(pfd::PowerFlowData) = pfd.lcc.rectifier_tap
+get_lcc_inverter_tap(pfd::PowerFlowData) = pfd.lcc.inverter_tap
+get_lcc_rectifier_delay_angle(pfd::PowerFlowData) = pfd.lcc.rectifier_delay_angle
+get_lcc_inverter_extinction_angle(pfd::PowerFlowData) = pfd.lcc.inverter_extinction_angle
+get_lcc_rectifier_phi(pfd::PowerFlowData) = pfd.lcc.rectifier_phi
+get_lcc_inverter_phi(pfd::PowerFlowData) = pfd.lcc.inverter_phi
+get_lcc_rectifier_bus(pfd::PowerFlowData) = pfd.lcc.rectifier_bus
+get_lcc_inverter_bus(pfd::PowerFlowData) = pfd.lcc.inverter_bus
+get_lcc_rectifier_transformer_reactance(pfd::PowerFlowData) =
+    pfd.lcc.rectifier_transformer_reactance
+get_lcc_inverter_transformer_reactance(pfd::PowerFlowData) =
+    pfd.lcc.inverter_transformer_reactance
+get_lcc_rectifier_min_alpha(pfd::PowerFlowData) = pfd.lcc.rectifier_min_alpha
+get_lcc_inverter_min_gamma(pfd::PowerFlowData) = pfd.lcc.inverter_min_gamma
+get_lcc_rectifier_i_dc(pfd::PowerFlowData) = pfd.lcc.rectifier_i_dc
+get_lcc_inverter_i_dc(pfd::PowerFlowData) = pfd.lcc.inverter_i_dc
+
+function LCCParameters(
+    sys::PSY.System,
+    bus_lookup::Dict{Int, Int},
+    reverse_bus_search_map::Dict{Int, Int},
+    time_steps::Int,
+)
+    lccs = PSY.get_components(PSY.get_available, PSY.TwoTerminalLCCLine, sys)
+    n_lccs = length(lccs)
+    lcc_p_set = zeros(Float64, n_lccs, time_steps)
+    lcc_rectifier_i_dc = zeros(Float64, n_lccs, time_steps)
+    lcc_inverter_i_dc = zeros(Float64, n_lccs, time_steps)
+    lcc_dc_line_resistance = zeros(Float64, n_lccs)
+    lcc_rectifier_tap = ones(Float64, n_lccs, time_steps)
+    lcc_inverter_tap = ones(Float64, n_lccs, time_steps)
+    lcc_rectifier_delay_angle = zeros(Float64, n_lccs, time_steps)
+    lcc_inverter_extinction_angle = zeros(Float64, n_lccs, time_steps)
+    lcc_rectifier_phi = zeros(Float64, n_lccs, time_steps)
+    lcc_inverter_phi = zeros(Float64, n_lccs, time_steps)
+    lcc_rectifier_bus = zeros(Int, n_lccs)
+    lcc_inverter_bus = zeros(Int, n_lccs)
+    lcc_rectifier_transformer_reactance = zeros(Float64, n_lccs)
+    lcc_inverter_transformer_reactance = zeros(Float64, n_lccs)
+    lcc_rectifier_min_alpha = zeros(Float64, n_lccs)
+    lcc_inverter_min_gamma = zeros(Float64, n_lccs)
+
+    lccs = PSY.get_components(PSY.get_available, PSY.TwoTerminalLCCLine, sys)
+    _get_lcc_bus_ix =
+        (bus) -> _get_bus_ix(bus_lookup, reverse_bus_search_map, PSY.get_number(bus))
```

Comment:
I admire the attempt to make more elegant but given that it's only used twice I'd say we should just call `_get_bus_ix` directly rather than incurring the overhead of this dynamically defined function